consent-ws/src/main/java/org/broadinstitute/consent/http/service/DatabaseDataSetAPI.java

```diff
@@ -195,127 +239,212 @@ public ParseResult overwrite(File dataSetFile, Integer userId) {
     }
 
     @Override
-    public  List< Map<String, String>> autoCompleteDataSets(String partial) {
+    public List<Map<String, String>> autoCompleteDataSets(String partial) {
         return dsDAO.getObjectIdsbyPartial(partial);
     }
 
     @Override
-    public void deleteDataset(String datasetObjectId, Integer dacUserId) throws IllegalStateException{
+    public List<Map<String, String>> getCompleteDataSet(String dataSetName) {
+        return dsDAO.getObjectIdsbyDataSetName(dataSetName);
+    }
+    @Override
+    public void deleteDataset(Integer dataSetId, Integer dacUserId) throws IllegalStateException {
         try {
             dsDAO.begin();
             dataSetAuditDAO.begin();
-            DataSet dataset = dsDAO.findDataSetByObjectId(datasetObjectId);
-            Collection<Integer> dataSetId = new ArrayList<>();
-            dataSetId.add(dataset.getDataSetId());
-            if(checkDatasetExistence(dataset.getDataSetId())){
+            DataSet dataset = dsDAO.findDataSetById(dataSetId);
+            Collection<Integer> dataSetsId = new ArrayList<>();
+            dataSetsId.add(dataset.getDataSetId());
```

Comment:
Could be combined with line above for a one-liner
```suggestion
            Collection<Integer> dataSetsId = Collections.singletonList(dataset.getDataSetId());
```